### DISCUSSION TITLE
本地编译参考文献报错

### DISCUSSION THREAD

**Discussion by Ishaan-Yang** [nju-lug/NJUThesis#122] [Q&A]
在NJU在线平台上正常编译，但在本地 Miktex+texworks下 xelatex+MakeIndex+Bibtex 编译失败。
所有参考文献均显示诸如 ` LaTeX Warning: Citation 'science.1069571' on page 1 undefined on input line 8.` 的错误。编译好的pdf文件中没有参考文献页面，`\cite{}`处均变成诸如
![Screenshot 2022-04-09 165521](https://user-images.githubusercontent.com/78727052/162564513-7a8b12ee-8711-4cdf-8923-562f45a8368d.png)
的形态。
这是为什么呢？

**Comment by atxy-blip** [comment]
1. `njuthesis` 依赖的参考文献工具是 `biber` 而非 `BibTeX`，这是因为后者出现较早，对于部分特殊情况的支持不好。所以，为了抛弃历史包袱，就没做 `BibTeX` 的支持，改用 `biblatex` 宏包加 `biber` 后端的方案。

2. 默认情况下也许并不需要 `makeindex` 工具生成附录。
